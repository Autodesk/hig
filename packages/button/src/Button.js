import React, { Component } from "react";
import PropTypes from "prop-types";
import { ControlBehavior } from "@hig/behaviors";

import {
  availableTargets,
  availableTypes,
  availableWidths,
  types,
  widths
} from "./constants";
import ButtonPresenter from "./presenters/ButtonPresenter";

export default class Button extends Component {
  static propTypes = {
    /**
     * Prevents user interaction with the button
     */
    disabled: PropTypes.bool,
    /**
     * A @hig/icon element
     */
    icon: PropTypes.node,
    /**
     * Sets the link of a button
     */
    link: PropTypes.string,
    /**
     * Triggers blur when focus is moved away from icon
     */
    onBlur: PropTypes.func,
    /**
     * Triggers when you click the button
     */
    onClick: PropTypes.func,
    /**
     * Triggers when focus is moved to button
     */
    onFocus: PropTypes.func,
    /**
     * Triggers when you hover over the button
     */
    onHover: PropTypes.func,
    /**
     * Triggers when the user's mouse is pressed over the button
     */
    onMouseDown: PropTypes.func,
    /**
     * Triggers when the user's mouse is over the button
     */
    onMouseEnter: PropTypes.func,
    /**
     * Triggers when the user's mouse is no longer over the button
     */
    onMouseLeave: PropTypes.func,
    /**
     * Triggers when the user's mouse is no longer pressed over the button
     */
    onMouseUp: PropTypes.func,
    /**
     * Adds custom/overriding styles
     */
    stylesheet: PropTypes.func,
    /**
     * Specifies where to display the linked URL
     */
    target: PropTypes.oneOf(availableTargets),
    /**
     * Sets the title of a button
     */
    title: PropTypes.string.isRequired,
    /**
     * Specifies type of button
     */
    type: PropTypes.oneOf(availableTypes),
    /**
     * Specifies width of button
     */
    width: PropTypes.oneOf(availableWidths)
  };

  static defaultProps = {
    disabled: false,
    type: types.SOLID,
    width: widths.SHRINK
  };

  render() {
    const {
      disabled,
      icon,
      link,
      onBlur,
      onClick,
      onFocus,
      onHover,
      onMouseDown,
      onMouseEnter,
      onMouseLeave,
      onMouseUp,
      stylesheet,
      target,
      title,
      type,
      width,
      ...otherProps
    } = this.props;

    return (
      <ControlBehavior
        onBlur={onBlur}
        onFocus={onFocus}
        onMouseDown={onMouseDown}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
        onMouseUp={onMouseUp}
      >
        {({
          hasFocus,
          hasHover,
          isPressed,
          onBlur: handleBlur,
          onFocus: handleFocus,
          onMouseDown: handleMouseDown,
          onMouseEnter: handleMouseEnter,
          onMouseLeave: handleMouseLeave,
          onMouseUp: handleMouseUp
        }) => (
          <ButtonPresenter
            disabled={disabled}
            hasFocus={hasFocus}
            hasHover={hasHover}
            isPressed={isPressed}
            icon={icon}
            link={link}
            onBlur={handleBlur}
            onClick={onClick}
            onFocus={handleFocus}
            onHover={onHover}
            onMouseDown={handleMouseDown}
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
            onMouseUp={handleMouseUp}
            stylesheet={stylesheet}
            target={target}
            title={title}
            type={type}
            width={width}
            {...otherProps}
          />
        )}
      </ControlBehavior>
    );
  }
}
