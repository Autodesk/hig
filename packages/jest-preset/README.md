# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
