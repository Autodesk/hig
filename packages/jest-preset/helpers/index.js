/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
