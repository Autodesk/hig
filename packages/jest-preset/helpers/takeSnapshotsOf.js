const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
