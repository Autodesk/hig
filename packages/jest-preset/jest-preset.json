{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
