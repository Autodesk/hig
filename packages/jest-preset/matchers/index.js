require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
