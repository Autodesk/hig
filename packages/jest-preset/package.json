{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
