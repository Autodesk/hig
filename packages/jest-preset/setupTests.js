require("jest-enzyme");
require("jest-extended");
require("./matchers");
