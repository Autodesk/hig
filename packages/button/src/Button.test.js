import React from "react";
import { mount } from "enzyme";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import Button from "./Button";

describe("Button", () => {
  const subject = (props = {}) => mount(<Button title="Button" {...props} />);
  let wrapper;

  takeSnapshotsOf(Button, [
    {
      description: "renders as default button",
      props: { title: "Settings" }
    },
    {
      description: "renders as button type solid",
      props: { title: "Settings", type: "solid" }
    },
    {
      description: "renders as button type outline",
      props: { title: "Settings", type: "outline" }
    },
    {
      description: "renders as button type flat",
      props: { title: "Settings", type: "flat" }
    },
    {
      description: "renders with an icon",
      props: { icon: "<Icon />", title: "Settings" }
    },
    {
      description: "renders with className prop",
      props: { title: "Button Class", className: "custom-class" }
    },
    {
      description: "renders with a link",
      props: { link: "https://hig.autodesk.com", title: "Settings" }
    }
  ]);

  describe("event handlers", () => {
    let eventHandler;

    describe("onBlur", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onBlur: eventHandler });
      });

      it("is triggered on blur", () => {
        wrapper.simulate("blur");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onClick", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onClick: eventHandler });
      });

      it("is triggered on click", () => {
        wrapper.simulate("click");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onFocus", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onFocus: eventHandler });
      });

      it("is triggered on focus", () => {
        wrapper.simulate("focus");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onHover", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onHover: eventHandler });
      });

      it("is triggered on mouseover", () => {
        wrapper.simulate("mouseover");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onMouseEnter", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onMouseEnter: eventHandler });
      });

      it("is triggered on mouse enter", () => {
        wrapper.simulate("mouseenter");
        expect(eventHandler).toBeCalled();
      });
    });

    describe("onMouseLeave", () => {
      beforeEach(() => {
        eventHandler = jest.fn();
        wrapper = subject({ onMouseLeave: eventHandler });
      });

      it("is triggered on mouse leave", () => {
        wrapper.simulate("mouseleave");
        expect(eventHandler).toBeCalled();
      });
    });
  });

  describe("the element rendered", () => {
    let link;
    describe("given a link", () => {
      beforeEach(() => {
        link = "https://www.autodesk.com";
        wrapper = subject({ link });
      });

      it("is an <a> tag", () => {
        expect(wrapper.matchesElement(<a href={link}>Button</a>));
      });
    });

    describe("without a link", () => {
      beforeEach(() => {
        wrapper = subject();
      });

      it("is a <button> tag", () => {
        expect(wrapper.matchesElement(<button>Button</button>));
      });
    });
  });

  describe("when a disabled link", () => {
    const link = "https://www.autodesk.com";

    beforeEach(() => {
      wrapper = subject({ link, disabled: true });
    });

    it("is not tabbable", () => {
      expect(
        wrapper.matchesElement(
          <a href={link} tabIndex="-1">
            Button
          </a>
        )
      );
    });
  });
});
