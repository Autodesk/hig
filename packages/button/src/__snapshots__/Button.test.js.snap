// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  1`] = `
.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: #0696d7;
}

.emotion-0:visited {
  color: #ffffff;
}

.emotion-0 svg * {
  fill: #ffffff;
}

<button
  className="emotion-0"
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className=""
  >
    Settings
  </span>
</button>
`;

exports[`Button  2`] = `
.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: #0696d7;
}

.emotion-0:visited {
  color: #ffffff;
}

.emotion-0 svg * {
  fill: #ffffff;
}

<button
  className="emotion-0"
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className=""
  >
    Settings
  </span>
</button>
`;

exports[`Button  3`] = `
.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: rgba(128,128,128,0.5);
  border-radius: 2px;
  box-sizing: border-box;
  color: #3c3c3c;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: none;
}

.emotion-0:visited {
  color: #3c3c3c;
}

.emotion-0 svg * {
  fill: #808080;
}

<button
  className="emotion-0"
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className=""
  >
    Settings
  </span>
</button>
`;

exports[`Button  4`] = `
.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: rgba(128,128,128,0);
  border-radius: 2px;
  box-sizing: border-box;
  color: #006EAF;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: none;
}

.emotion-0:visited {
  color: #006EAF;
}

.emotion-0 svg * {
  fill: #006EAF;
}

<button
  className="emotion-0"
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className=""
  >
    Settings
  </span>
</button>
`;

exports[`Button  5`] = `
.emotion-2 {
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: #0696d7;
}

.emotion-2:visited {
  color: #ffffff;
}

.emotion-2 svg * {
  fill: #ffffff;
}

.emotion-0 {
  display: inline-block;
  position: absolute;
  height: 24px;
  margin-right: 4px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-1 {
  margin-left: calc(24px + 4px);
}

<button
  className="emotion-2"
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className="emotion-0"
  >
    &lt;Icon /&gt;
  </span>
  <span
    className="emotion-1"
  >
    Settings
  </span>
</button>
`;

exports[`Button  6`] = `
.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: #0696d7;
}

.emotion-0:visited {
  color: #ffffff;
}

.emotion-0 svg * {
  fill: #ffffff;
}

<button
  className="emotion-0 custom-class"
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className=""
  >
    Button Class
  </span>
</button>
`;

exports[`Button  7`] = `
.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: #0696d7;
}

.emotion-0:visited {
  color: #ffffff;
}

.emotion-0 svg * {
  fill: #ffffff;
}

<a
  className="emotion-0"
  disabled={false}
  href="https://hig.autodesk.com"
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={undefined}
  onMouseUp={[Function]}
  tabIndex="0"
  target={undefined}
>
  <span
    className=""
  >
    Settings
  </span>
</a>
`;
